File: python/fatorial.py

def iterativo(n):
    res = n
    for i in range(2, n):
        res *= i
    return res

def recursivo(n):
    return 1 if n < 2 else n * recursivo(n-1)

from operator import mul

def funcional(n):
    return reduce(mul, xrange(1, n+1))

File: ruby/fatorial.rb

def iterativo(n)
    f = 1
    for i in 1..n
        f *= i
    end
    f
end

def recursivo(n)
    n < 2 ? 1 : n * recursivo(n-1)
end

def funcional(n)
    (1..n).reduce(:*)
end
